### DISCUSSION TITLE
How to setup giscus login with my website login system in a next.js project?

### DISCUSSION THREAD

**Discussion by alaminsahed** [giscus/giscus#1001] [Q&A]
I Integrated "[giscus](https://github.com/giscus/giscus-component)" in my next.js project. My project already has a GitHub login system (using nextAuth).  I don't want the user to log in again to use giscus. How can I set up giscus with my website auth system? That means users only log in to my website and can comment without login again.

**Comment by kakduwik** [comment]
Hi @alaminsahed do you able to do it?

**Comment by EARL146** [comment]
Giscus uses GitHub Discussions authentication, so it does not directly connect to your website login system.

For a Next.js project, you can manage your own users with NextAuth/Auth.js, but giscus comments still require GitHub authentication.

You can sync the theme or UI with your Next.js app, but the authentication flow is handled by GitHub.
